### PR TITLE
Fix operator revision handling

### PR DESCRIPTION
**Pull request by ostromart**
A different logic is required to support separate gateway upgrade. See https://github.com/istio/istio.io/pull/8299 for the recommended way to manage gateways separately, which requires this change.
The effect of the change is that operator will:
- ignore any IstioOperator CR whose revision does not match its own
- prune resources within a revision based on the CR name. This way istiod and GW installs do not affect each other.

### CODE REVIEW

**Review by richardwxn — APPROVED** [review]

File: operator/pkg/helmreconciler/reconciler.go [hunk 1]
```diff
@@ -101,9 +101,6 @@ func NewHelmReconciler(client client.Client, restConfig *rest.Config, iop *value
 		iop = &valuesv1alpha1.IstioOperator{}
 		iop.Spec = &v1alpha1.IstioOperatorSpec{}
 	}
-	if operatorRevision, found := os.LookupEnv("REVISION"); found {
-		iop.Spec.Revision = operatorRevision
-	}
```
Comment (morvencao) [inline operator/pkg/helmreconciler/reconciler.go:106]: Does canary upgrade with in-cluster operator https://istio.io/latest/docs/setup/install/operator/#canary-upgrade still work if we remove this?

**Review by howardjohn — APPROVED** [review]

File: operator/pkg/helmreconciler/reconciler.go (hunk 1 above)
Comment (ostromart) [reply operator/pkg/helmreconciler/reconciler.go:112]: Yes, but the workflow is modified slightly: https://github.com/istio/istio.io/pull/8299

**Pull request merged**